### ISSUE TITLE
Installing jupyterlab 3.6.x on linux/arm64 gives an error about Cargo not being installed

### ISSUE THREAD

**Issue by benz0li** [jupyterlab/jupyterlab#13920]
## Description

jupyterlab 3.6.x depends on jupyter-server-ydoc<0.7.0,>=0.6.0 depends on {jupyter-ydoc~=0.2.2, ypy-websocket<0.9.0,>=0.8.2} depends on y-py<0.6.0,>=0.5.3.

For y-py there is no wheel available for `linux/arm64` (`aarch64`) and therefore requires Rust and Cargo to compile.

## Reproduce

```bash
pip install jupyterlab==3.6.1
```

```bash
Collecting jupyterlab==3.6.1
  Downloading jupyterlab-3.6.1-py3-none-any.whl (8.9 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 8.9/8.9 MB 12.5 MB/s eta 0:00:00
Collecting jupyterlab-server~=2.19
  Downloading jupyterlab_server-2.19.0-py3-none-any.whl (56 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 56.4/56.4 kB 4.2 MB/s eta 0:00:00
Collecting packaging
  Downloading packaging-23.0-py3-none-any.whl (42 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.7/42.7 kB 12.8 MB/s eta 0:00:00
Collecting jupyter-core
  Downloading jupyter_core-5.2.0-py3-none-any.whl (94 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 94.3/94.3 kB 26.8 MB/s eta 0:00:00
Collecting nbclassic
  Downloading nbclassic-0.5.1-py3-none-any.whl (10.0 MB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 10.0/10.0 MB 20.6 MB/s eta 0:00:00
Collecting jupyter-ydoc~=0.2.2
  Downloading jupyter_ydoc-0.2.2-py3-none-any.whl (5.6 kB)
Collecting jupyter-server<3,>=1.16.0
  Downloading jupyter_server-2.2.1-py3-none-any.whl (365 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 365.3/365.3 kB 14.0 MB/s eta 0:00:00
Collecting ipython
  Downloading ipython-8.9.0-py3-none-any.whl (783 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 783.1/783.1 kB 26.5 MB/s eta 0:00:00
Collecting notebook<7
  Downloading notebook-6.5.2-py3-none-any.whl (439 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 439.1/439.1 kB 18.4 MB/s eta 0:00:00
Collecting jinja2>=2.1
  Downloading Jinja2-3.1.2-py3-none-any.whl (133 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.1/133.1 kB 23.3 MB/s eta 0:00:00
Collecting tomli
  Downloading tomli-2.0.1-py3-none-any.whl (12 kB)
Collecting tornado>=6.1.0
  Downloading tornado-6.2-cp37-abi3-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (424 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 424.4/424.4 kB 31.8 MB/s eta 0:00:00
Collecting jupyter-server-ydoc<0.7.0,>=0.6.0
  Downloading jupyter_server_ydoc-0.6.1-py3-none-any.whl (11 kB)
Collecting MarkupSafe>=2.0
  Downloading MarkupSafe-2.1.2-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (26 kB)
Collecting jupyter-events>=0.4.0
  Downloading jupyter_events-0.6.3-py3-none-any.whl (18 kB)
Collecting jupyter-client>=7.4.4
  Downloading jupyter_client-8.0.2-py3-none-any.whl (103 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 103.3/103.3 kB 7.9 MB/s eta 0:00:00
Collecting websocket-client
  Downloading websocket_client-1.5.0-py3-none-any.whl (55 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.9/55.9 kB 10.1 MB/s eta 0:00:00
Collecting anyio>=3.1.0
  Downloading anyio-3.6.2-py3-none-any.whl (80 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.6/80.6 kB 18.0 MB/s eta 0:00:00
Collecting terminado>=0.8.3
  Downloading terminado-0.17.1-py3-none-any.whl (17 kB)
Collecting argon2-cffi
  Downloading argon2_cffi-21.3.0-py3-none-any.whl (14 kB)
Collecting jupyter-server-terminals
  Downloading jupyter_server_terminals-0.4.4-py3-none-any.whl (13 kB)
Collecting nbconvert>=6.4.4
  Downloading nbconvert-7.2.9-py3-none-any.whl (274 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 274.9/274.9 kB 25.9 MB/s eta 0:00:00
Collecting send2trash
  Downloading Send2Trash-1.8.0-py3-none-any.whl (18 kB)
Collecting prometheus-client
  Downloading prometheus_client-0.16.0-py3-none-any.whl (122 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 122.5/122.5 kB 27.8 MB/s eta 0:00:00
Collecting nbformat>=5.3.0
  Downloading nbformat-5.7.3-py3-none-any.whl (78 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 78.1/78.1 kB 15.1 MB/s eta 0:00:00
Collecting traitlets>=5.6.0
  Downloading traitlets-5.9.0-py3-none-any.whl (117 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 117.4/117.4 kB 28.1 MB/s eta 0:00:00
Collecting pyzmq>=24
  Downloading pyzmq-25.0.0-cp310-cp310-manylinux_2_17_aarch64.manylinux2014_aarch64.whl (853 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 853.7/853.7 kB 26.8 MB/s eta 0:00:00
Collecting platformdirs>=2.5
  Downloading platformdirs-2.6.2-py3-none-any.whl (14 kB)
Collecting ypy-websocket<0.9.0,>=0.8.2
  Downloading ypy_websocket-0.8.2-py3-none-any.whl (10 kB)
Collecting jupyter-server-fileid<1,>=0.6.0
  Downloading jupyter_server_fileid-0.6.0-py3-none-any.whl (15 kB)
Collecting y-py<0.6.0,>=0.5.3
  Downloading y_py-0.5.5.tar.gz (50 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 50.3/50.3 kB 14.4 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      
      Cargo, the Rust package manager, is not installed or is not on PATH.
      This package requires Rust and Cargo to compile extensions. Install it through
      the system's package manager or via https://rustup.rs/
      
      Checking for Rust toolchain....
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

## Expected behavior

When introducing new dependencies – e.g. jupyter-server-ydoc – please make sure there are wheels available for `linux/arm64` (`aarch64`).

## Context

- Operating System and version: Debian 11
- Browser and version: Not relevant
- JupyterLab version: 3.6.x

---

Ping @mathbunnyru: This also affects the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks).

**Comment by benz0li** [comment]
@Waidhoferj @kafonek Any chance of prividing wheels for `linux/arm64` in the foreseeable future?

**Comment by hbcarlos** [comment]
@davidbrochart

**Comment by bvmcode** [comment]
Same issue, had to install 3.5

**Comment by davidbrochart** [comment]
@benz0li do you want to open the issue in [ypy](https://github.com/y-crdt/ypy/issues/new)?

**Comment by benz0li** [comment]
> @benz0li do you want to open the issue in [ypy](https://github.com/y-crdt/ypy/issues/new)?

Done.

**Comment by mathbunnyru** [comment]
> Ping @mathbunnyru: This also affects the [Jupyter Docker Stacks](https://github.com/jupyter/docker-stacks).

You're right. There is actually automatic PR to make it work for conda-forge.
If that succeeds, it will work with mamba/conda (with conda-forge added).
https://github.com/conda-forge/y-py-feedstock/pull/16

**Comment by gecomagenta** [comment]
Same issue on a M1 mac running macOS Ventura 13.1 and python 3.9. 
Had to install jupyterlab version 3.5

**Comment by mathbunnyru** [comment]
It seems to work for conda-forge.
Hope this will be merged soon.

https://github.com/conda-forge/y-py-feedstock/pull/18

**Comment by davidbrochart** [comment]
> conda-forge/y-py-feedstock#18

It's merged.

**Comment by benz0li** [comment]
Resolved by https://github.com/y-crdt/ypy/commit/dd5b8eca66edcb915af98bcab55c8af19a1969bd (y-py 0.6.0), which provides a wheel for `linux/arm64` (`aarch64`).

---

@davidbrochart Why is there https://pypi.org/project/y-py/0.5.9/?

<img width="645" alt="y-py_0 5 5_release-date" src="https://user-images.githubusercontent.com/20233085/220550984-82900786-0e0c-42fc-ba00-423d8355b4aa.png">

**Comment by davidbrochart** [comment]
> @davidbrochart Why is there https://pypi.org/project/y-py/0.5.9/?

v0.6.0 has breaking changes, and v0.5.5 had inconsistent wheels, so I released a clean v0.5.9.

**Comment by benz0li** [comment]
> v0.6.0 has breaking changes, and v0.5.5 had inconsistent wheels, so I released a clean v0.5.9.

Makes sense. jupyterlab 3.6.x limits to y-py<0.6.0,>=0.5.3 and uses y-py 0.5.9.
